qa/tasks/mixed_system_tests/ios.py

```diff
@@ -0,0 +1,145 @@
+"""
+Component IOs
+"""
+
+import yaml
+import logging
+import os
+import pwd
+import time
+
+from teuthology.orchestra import run
+
+log = logging.getLogger(__name__)
+
+
+cephqe_scripts = {
+    "DIR": {"v1":
+                {"script": "rgw/v1/tests/s3/",
+                 "config": "rgw/v1/tests/s3/yamls"},
+            "v2":
+                {"script": "rgw/v2/tests/s3_swift/",
+                 "config": "rgw/v2/tests/s3_swift/configs"}
+            },
+    "MASTER_BRANCH": "master",
+    "REPO_NAME": "ceph-qe-scripts",
+    "WIP_BRANCH": None
+}
+
+
+class rgw_ios:
+    """
+    RGW IOS using ceph-qe-scripts
+    """
+
+    def __init__(self, ctx, config):
+        self.ctx = ctx
+        self.config = config
+
+    def __enter__(self):
+        log.info('starting rgw-tests')
+        log.info('config %s' % self.config)
+        if self.config is None:
+            self.config = {}
+        assert isinstance(self.config, dict), \
+            "task set-repo only supports a dictionary for configuration"
+        config_file_name = self.config['test'] + ".yaml"
+        log.info('test_version: %s' % self.config.get('test_version', 'v2'))
+        log.info('test: %s' % self.config['test'])
+        log.info('script: %s' % self.config.get('script', self.config['test'] + ".py"))
+        time.sleep(2)
+        test_root_dir = self.config['test'] + "_%d" % "_".join(str(time.time()).split("."))
+        test_venv = os.path.join(test_root_dir, "venv")
+        script = os.path.join(cephqe_scripts['REPO_NAME'],
+                              cephqe_scripts['DIR'][self.config.get('test_version', 'v2')]['script'],
+                              self.config.get('script', self.config['test'] + ".py"))
+        config_file = os.path.join(cephqe_scripts['REPO_NAME'],
+                                   cephqe_scripts['DIR'][self.config.get('test_version', 'v2')]['config'],
+                                   config_file_name)
+        log.info('script: %s' % script)
+        log.info('config_file: %s' % config_file)
+        self.soot = [test_venv, test_root_dir, 'io_info.yaml', '*.json', 'Download.*',
+                     'Download', '*.mpFile', 'x*', 'key.*', 'Mp.*', '*.key.*']
+        self.cleanup = lambda x: remote.run(args=[run.Raw('sudo rm -rf %s' % x)])
+        log.info('listing all clients: %s' % self.config.get('clients'))
+        for role in self.config.get('clients', ['client.0']):
```

Comment:
Are you cloning into every client, serially? Why not clone once and scp to all of them? Not to mention that half of this code could have been written more easily in Ansible?